src/framework/global/runtime.cpp

```diff
@@ -22,11 +22,24 @@
 
 #include "runtime.h"
 
+#ifdef Q_OS_LINUX
+#include <pthread.h>
+#endif
+
 static thread_local std::string s_threadName;
 
 void muse::runtime::setThreadName(const std::string& name)
 {
     s_threadName = name;
+#ifdef Q_OS_LINUX
+    // Set thread name through pthreads to aid debuggers that display such names.
+    // Thread names are limited to 16 bytes on Linux, including the
+    // terminating null.
+    std::string truncated_name = name.length() > 15 ? name.substr(0, 15) : name;
```

Comment:
I think we should add an assert that checks the length of the name:

`DO_ASSERT(name. length() <= 15);`